workflows/prognostic_c48_run/emulation/emulation/monitor.py

```diff
@@ -0,0 +1,242 @@
+import logging
```

Comment:
I think this module is sufficiently complex that it should have some unit tests. To simplify testing, instead of using globals, you could refactor it into a singleton class e.g.:
```
singleton = Class.from_environment(sys.environ)
store = singleton.store
...
```